pandas/tests/copy_view/test_methods.py

```diff
@@ -77,22 +77,31 @@ def test_reset_index(using_copy_on_write):
     tm.assert_frame_equal(df, df_orig)
 
 
-def test_rename_columns(using_copy_on_write):
+@pytest.mark.parametrize("copy", [True, None, False])
+def test_rename_columns(using_copy_on_write, copy):
     # Case: renaming columns returns a new dataframe
     # + afterwards modifying the result
     df = DataFrame({"a": [1, 2, 3], "b": [4, 5, 6], "c": [0.1, 0.2, 0.3]})
     df_orig = df.copy()
-    df2 = df.rename(columns=str.upper)
+    df2 = df.rename(columns=str.upper, copy=copy)
 
-    if using_copy_on_write:
+    if (using_copy_on_write and copy is not True) or copy is False:
```

Comment:
The current PR experiments with combining testing the `copy=True/False/None` keyword in those tests that we have been adding to test the CoW behaviour of those methods. 

Combining it would give the least duplication, but, I have to say that the `if` expressions are getting quite complicated and hard to follow .. 
For example here `if (using_copy_on_write and copy is not True) or copy is False` and below `if using_copy_on_write or copy is not False` and `if using_copy_on_write and copy is not True`, all slightly different.